src/components/Extension/PrivateExtensions/actions/IntakeScheduling/api.service.ts

```diff
@@ -22,39 +25,74 @@ export const fetchProviders = async (
     })
 
     if (!response.ok) {
+      log(
+        {
+          message: `${basicMessage}: failed`,
+          data: { input, response },
+        },
+        'ERROR',
+        ''
+      )
       throw new Error('Failed to fetch providers')
     }
 
     const jsonRes = await response.json()
     const result = GetProvidersResponseSchema.safeParse(jsonRes)
 
     if (!result.success) {
-      console.error('Zod parsing error', result.error.issues)
+      log(
+        {
+          message: `${basicMessage}: zod parsing error`,
```

Comment:
This error deserves some special attention because it means either Awell or SOL violated the API contract. I.e. the payload we received doesn't match the payload we agreed upon.

So I'd maybe not label it as a Zod error but call it something else and give it a more explicit name (maybe it's own error class?)

Example:
```
class SolApiResponseError extends Error {
  public issues

  constructor(message: string, issues: any[]) {
    super(message);
    this.name = 'SolApiResponseError';
    this.issues = issues;
  }
}
```